sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -73,19 +73,21 @@ case class UserDefinedFunction protected[sql] (
    */
   @scala.annotation.varargs
   def apply(exprs: Column*): Column = {
-    if (inputTypes.isDefined && nullableTypes.isDefined) {
-      require(inputTypes.get.length == nullableTypes.get.length)
+    val numOfArgs = ScalaReflection.getParameterCount(f)
```

Comment:
since we are here, shall we also check `exprs.length == numOfArgs`?

Reply:
Actually I think the "assert" should better be in `ScalaUDF`, but then it would be slightly different from the current behavior whatsoever. I'm keeping other things as they are.